fili-generic-example/src/main/java/com/yahoo/wiki/webservice/data/config/auto/TableConfig.java

```diff
@@ -122,7 +122,7 @@ public TimeGrain getValidTimeGrain() {
     }
 
     @Override
-    public List<DataSegment> getDataSegmentMetadata() {
-        return dataSegments;
+    public List<DataSegment> getDataSegments() {
+        return Collections.unmodifiableList(dataSegments);
```

Comment:
Other getters in this class are all returning immutables. It also make sense to do the same here.